src/Behat/Symfony2Extension/Context/Argument/ServiceArgumentResolver.php

```diff
@@ -108,15 +112,18 @@ private function getServiceName($argument)
     }
 
     /**
-     * @param ContainerInterface $container
-     * @param string $argument
-     * @return string
+     * Sanitise the container key given by the Behat config file,
+     * and retrieve the parameter from the Container.
+     *
+     * @param  ContainerInterface $container
+     * @param  string $key
+     * @return mixed
      */
-    private function replaceParameters(ContainerInterface $container, $argument)
+    private function replaceParameters(ContainerInterface $container, $key)
     {
-        return preg_replace_callback('/(?<!%)%([^%]+)%(?!%)/', function($matches) use ($container) {
-            return $container->getParameter($matches[1]);
-        }, $argument);
+        $cleanKey = preg_replace('/(?<!%)%([^%]+)%(?!%)/', '$1', $key);
```

Comment:
Good point, as explained in my comment below, `preg_replace` worked for some cases, except I needed to check if there was a match before I proceeded. I have updated to use `preg_match`.